cmd/renterd/main.go

```diff
@@ -102,9 +102,58 @@ func flagCurrencyVar(c *types.Currency, name string, d types.Currency, usage str
 	flag.Var(newCurrencyVar(c, d), name, usage)
 }
 
+type (
```

Comment:
To me a heap feels like quite the overcomplicated solution. Especially since we pretty much want to close things in reverse order of creating them anyway right?

So we can do
```
log.Println("Shutting down...")
if autopilotCleanup != nil {
  autopilotCleanup()
}
 srv.Shutdown(context.Background())
```
to shut down the autopilot first, then the server and finally all the already deferred cleanups.

It's not a big deal to seemingly miss the cleanup when an error occurs because we use `log.Fatal` to report the error anyway. Which just kills the process without executing defers.

Reply:
Yeah went back and forth a couple of times. I removed it and then added it back. It's the listener that made me re-add it.. the listener is created first and thus the close function is called last. 

I figured the order should be:
- autopilot
- server
- listener
- worker
- bus
- log files
- tracing

I'll just remove it. I don't think it's that complicated though, it's already kind of a `defer` mess which is why I figured a heap isn't the worst thing